provider/core/src/request.rs

```diff
@@ -231,11 +248,18 @@ impl FromStr for DataLocale {
                 .with_display_context(s)
                 .with_display_context(&e)
         })?;
+        #[cfg_attr(not(feature = "experimental"), allow(unused_mut))]
         let mut data_locale = DataLocale::from(locale);
+        #[cfg(feature = "experimental")]
         if let Some(aux_str) = aux_iter.next() {
             let aux = AuxiliaryKeys::from_str(aux_str)?;
             data_locale.set_aux(aux);
         }
+        if aux_iter.next().is_some() {
+            return Err(DataErrorKind::KeyLocaleSyntax
+                .into_error()
+                .with_display_context(s));
+        }
```

Comment:
Note: This code here is a behavior change (not allowing unknown characters after parsing), but this FromStr impl is new in this release. I added a test for it.